src/main/groovy/io/seqera/wave/controller/MetricsController.groovy

```diff
@@ -0,0 +1,168 @@
+/*
+ *  Wave, containers provisioning service
+ *  Copyright (c) 2024, Seqera Labs
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package io.seqera.wave.controller
+
+import java.time.Instant
+import java.time.LocalDate
+import java.time.LocalTime
+import java.time.ZoneId
+import java.time.format.DateTimeParseException
+import javax.annotation.Nullable
+
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import io.micronaut.http.HttpResponse
+import io.micronaut.http.MediaType
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Error
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.annotation.PathVariable
+import io.micronaut.http.annotation.QueryValue
+import io.micronaut.security.annotation.Secured
+import io.micronaut.security.authentication.AuthorizationException
+import io.micronaut.security.rules.SecurityRule
+import io.seqera.wave.service.metric.Metric
+import io.seqera.wave.service.metric.MetricFilter
+import io.seqera.wave.service.metric.MetricService
+import jakarta.inject.Inject
+import static io.micronaut.http.HttpHeaders.WWW_AUTHENTICATE
+
+/**
+ * Controller for wave Metrics
+ *
+ * @author Munish Chouhan <munish.chouhan@seqera.io>
+ */
+@Slf4j
+@CompileStatic
+@Secured(SecurityRule.IS_AUTHENTICATED)
+@Controller("/v1alpha1/metrics")
+class MetricsController {
+    @Inject
+    private MetricService metricsService
+
+    @Get(uri = "/builds/{metric}", produces = MediaType.APPLICATION_JSON)
+    HttpResponse<Map> getBuildMetrics(@PathVariable String metric,
+                                      @Nullable @QueryValue Boolean success,
+                                      @Nullable @QueryValue String startDate,
+                                      @Nullable @QueryValue String endDate,
+                                      @Nullable @QueryValue Integer limit) {
+        return HttpResponse.ok(
+                metricsService.getBuildMetrics(
+                        Metric.valueOf(metric),
+                        new MetricFilter.Builder()
+                                .dates(parseStartDate(startDate), parseEndDate(endDate))
+                                .success(success)
+                                .limit(limit)
+                                .build()
+                )
```

Comment:
Let's use a proper `GetBuildsMetricsResponse` object

Reply:
@pditommaso Do you mean something like this?
1. 
```
class BuildsMetricsResponse {
    Metric metric
    Map<String, Long> data
}
```
and
```
class BuildsMetricsCountResponse {
    Metric metric
    Long Count
}
```

2. Or should I use a list instead of Map?

```
class BuildsMetricsResponse {
    String name
    Long count
}
```
then return List<MetricResponse> and MetricResponse

Reply:
Sounds good

Reply:
@pditommaso Sorry do you mean the first option or the second?

Reply:
Likely the first, having `metric` as a string

Reply:
thank you

Reply:
Thinking more it should mimic the current result just wrapper into a response object. Therefore something like 

```
class GetBuildsMetricsResponse {
    Map<String, Long> result
}

class GetBuildsCountResponse {
    Long count
}
```

Not the structure <Verb><Target>`Response`

Reply:
ok understood

Reply:
done